internal/cache/target.go

```diff
@@ -117,26 +117,36 @@ type referenceTarget struct {
 	registry.ReferenceFetcher
 }
 
-// FetchReference fetches the content identified by the reference from the
-// remote and cache the fetched content.
+// FetchReference fetches the content identified by the reference from the cache
+// if exists, otherwise fetches the content from the remote and then cache the
+// fetched content.
 // Cached content will only be read via Fetch, FetchReference will always fetch
 // From origin.
 func (t *referenceTarget) FetchReference(ctx context.Context, reference string) (ocispec.Descriptor, io.ReadCloser, error) {
-	target, rc, err := t.ReferenceFetcher.FetchReference(ctx, reference)
+	desc, err := t.Resolve(ctx, reference)
 	if err != nil {
```

Comment:
And `errdef.ErrNotFound` should be regarded as not cached and fetch from origin.